### PR TITLE
Update development.md

### PR DESCRIPTION
**Pull request by Itsamanjain**
I've optimized  development.md for clarity and ease of understanding for beginner developers

### What does this PR do?

Simplified Language: The language used in the README was simplified to make it more accessible for beginner developers. Complex technical terms were explained or replaced with simpler terms where appropriate.

Improved Structure: The README was restructured to follow a logical order, making it easier for readers to follow the steps required to set up their development environment for Bun.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: docs/project/development.md [hunk 1]
```diff
@@ -1,293 +1,160 @@
...
 
-```bash#Native
-$ curl -fsSL https://bun.sh/install | bash # for macOS, Linux, and WSL
+Before starting, make sure you already have a release build of Bun installed because we'll use it to transpile and minify our code.
+
+## Installation Steps
+
+Choose one of the following methods to install Bun:
+
+### Method 1: Using curl (for macOS, Linux, and WSL)
```
Comment (paperclover) [inline docs/project/development.md:14]: please revert back to using the codetabs component. this looks worse

File: docs/project/development.md [hunk 2]
```diff
@@ -1,293 +1,160 @@
...
 ```
 
-Note: `make setup` is just an alias for the following:
+Note: `make setup` is equivalent to the following command:
 
 ```bash
 $ make assert-deps submodule npm-install-dev node-fallbacks runtime_js fallback_decoder bun_error mimalloc picohttp zlib boringssl libarchive lolhtml sqlite usockets uws tinycc c-ares zstd base64 cpp zig link
 ```
 
-## Rebuilding
```
Comment (paperclover) [inline docs/project/development.md:152]: why was this table removed. it is a required resource for rebuilding parts of bun, until we move to CMake. `make setup` is intended only to be run once because running it again rebuilds alot of code and is overall slower.

File: docs/project/development.md [hunk 3]
```diff
@@ -1,293 +1,160 @@
...
-The above will probably also need Zig and/or C++ code rebuilt.
+This command cloned submodules and built subprojects. When a submodule is out of date, you can run `make submodule` to quickly reset/update all your submodules and then rebuild individual submodules with their respective command.
 
 ## VSCode
 
-VSCode is the recommended IDE for working on Bun, as it has been configured. Once opening, you can run `Extensions: Show Recommended Extensions` to install the recommended extensions for Zig and C++. ZLS is automatically configured.
+VSCode is the recommended IDE for working on Bun, as it has been pre-configured. Upon opening, you can run `Extensions: Show Recommended Extensions` to install the recommended extensions for Zig and C++. ZLS (Zig Language Server) is automatically configured.
 
-### ZLS
+### ZLS (Zig Language Server)
```
Comment (paperclover) [inline docs/project/development.md:155]: not really harm in having the () in this header but is it really needed when the first sentence says what it is.

i think we can actually comment this whole ZLS section out because the latest verison works better than it did before.

File: docs/project/development.md [hunk 4]
```diff
@@ -305,69 +172,65 @@ Then add absolute paths to Zig and ZLS in your vscode config:
...
 ```
 
-## JavaScript builtins
+## JavaScript Builtins
 
-When you change anything in `src/js/builtins/*` or switch branches, run this:
+When you modify anything in `src/js/builtins/*` or switch branches, run this:
 
 ```bash
 $ make js cpp
```
Comment (paperclover) [inline docs/project/development.md:180]: my mistake, can you change this to `make cpp` because cpp implies rebundling js

File: docs/project/development.md [hunk 5]
```diff
@@ -305,69 +172,65 @@ Then add absolute paths to Zig and ZLS in your vscode config:
...
 ```
 
-That inlines the TypeScript code into C++ headers.
+This inlines the TypeScript code into C++ headers.
 
-{% callout %}
-Make sure you have `ccache` installed, otherwise regeneration will take much longer than it should.
```
Comment (paperclover) [inline docs/project/development.md:319]: why was this removed. this is an important reminder.

File: docs/project/development.md [hunk 6]
```diff
@@ -376,135 +239,84 @@ $ make release
 
 The binary will be located at `packages/bun-{platform}-{arch}/bun`.
 
-## Valgrind
-
-On Linux, valgrind can help find memory issues.
-
-Keep in mind:
+## Valgrind (Linux Only)
```
Comment (paperclover) [inline docs/project/development.md:242]: i dont think (Linux only) is needed to be clarified when it is clarified again in the first two words of the section.

File: docs/project/development.md [hunk 7]
```diff
@@ -305,69 +172,65 @@ Then add absolute paths to Zig and ZLS in your vscode config:
...
-```
+```bash
 $ make js
 ```
 
-In debug builds, Bun automatically loads these from the filesystem, wherever it was compiled, so no need to re-run `make dev`.
+In debug builds, Bun automatically loads these modules from the filesystem, so there's no need to rerun `make dev`.
 
-## Release build
+## Building a Release Version
```
Comment (paperclover) [inline docs/project/development.md:231]: imo "Release Build" was better vs "Release Version"

File: docs/project/development.md [hunk 8]
```diff
@@ -376,135 +239,84 @@ $ make release
...
-- Valgrind is slow
-- Mimalloc will sometimes cause spurious errors when debug build is enabled
+On Linux, Valgrind can help find memory issues. Keep in mind:
 
-You'll need a very recent version of Valgrind due to DWARF 5 debug symbols. You may need to manually compile Valgrind instead of using it from your Linux package manager.
+- JavaScriptCore doesn't support Valgrind and may report spurious errors.
+- Valgrind is slow.
+- Mimalloc may cause spurious errors when the debug build is enabled.
 
-`--fair-sched=try` is necessary if running multithreaded code in Bun (such as the bundler). Otherwise it will hang.
```
Comment (paperclover) [inline docs/project/development.md:391]: why was this removed?

File: docs/project/development.md [hunk 9]
```diff
@@ -376,135 +239,84 @@ $ make release
...
+Ensure you have a recent version of Valgrind with DWARF 5 debug symbols. You may need to compile Valgrind manually:
 
 ```bash
 $ valgrind --fair-sched=try --track-origins=yes bun-debug <args>
 ```
 
-## Updating `WebKit`
+## Updating WebKit
 
-The Bun team will occasionally bump the version of WebKit used in Bun. When this happens, you may see something like this with you run `git status`.
```
Comment (paperclover) [inline docs/project/development.md:399]: why was this removed.

File: docs/project/development.md [hunk 10]
```diff
@@ -376,135 +239,84 @@ $ make release
...
-$ cd src/bun.js/WebKit
-$ git fetch
-$ git checkout <hash>
-``` -->
+If you encounter issues, check the [Bun repo](https://github.com/oven-sh/bun/tree/main/src/bun.js) for the hash of the current WebKit commit in use.
 
 ## Troubleshooting
 
-### 'span' file not found on Ubuntu
+If you encounter issues during the setup process, here are some troubleshooting steps:
```
Comment (paperclover) [inline docs/project/development.md:269]: i dont think this line is needed, and it also implies these are only setup issues. i think that *is* the case right now but we may add some more of these in the future.

File: docs/project/development.md [hunk 11]
```diff
@@ -376,135 +239,84 @@ $ make release
...
-
-```bash
 $ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100
 $ sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
 ```
 
 ### libarchive
 
-If you see an error when compiling `libarchive`, run this:
+If you encounter issues with `libarchive` during compilation, run:
```
Comment (paperclover) [inline docs/project/development.md:284]: this is less clear. (prev specifies it's an error [message] about libarchive)

File: docs/project/development.md [hunk 12]
```diff
@@ -376,135 +239,84 @@ $ make release
...
 
 ```bash
 $ xcode-select --install
 ```
 
-## Arch Linux / Cannot find `libatomic.a`
+## Arch Linux - Cannot Find 'libatomic.a'
 
-Bun requires `libatomic` to be statically linked. On Arch Linux, it is only given as a shared library, but as a workaround you can symlink it to get the build working locally.
+On Arch Linux, Bun requires 'libatomic' to be statically linked. You can create a symlink to make it work locally:
```
Comment (paperclover) [inline docs/project/development.md:316]: why was the note about libatomic only being available as shared library removed.

**Review by paperclover — CHANGES_REQUESTED** [review]
There are good grammar and style fixes here, but it removes too much information about how to work on the Bun codebase; This cannot be merged in this state.

**Comment by Itsamanjain** [comment]
@paperdave  I'll make the changes that you mentioned above
